crates/ruff_linter/src/rules/pylint/rules/boolean_chained_comparison.rs

```diff
@@ -83,20 +91,44 @@ pub(crate) fn boolean_chained_comparison(checker: &mut Checker, expr_bool_op: &E
                 return None;
             }
 
-            let edit = Edit::range_replacement(
-                left_compare_right.id().to_string(),
-                TextRange::new(left_compare_right.start(), right_compare_left.end()),
+            let left_has_paren = parenthesized_range(
+                ExpressionRef::Compare(left_compare),
+                expr_bool_op.as_any_node_ref(),
```

Comment:
Nit: You can do the `into` game if you want to
```suggestion
                left_compare.into(),
                expr_bool_op.into(),
```